### PR TITLE
Add GitHub labels

### PR DESCRIPTION
**Pull request by paulacamargo25**
Closed: #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -462,6 +462,7 @@
     "ts-mockito": "^2.6.1",
     "typemoq": "^2.1.0",
     "typescript": "^4.7.4",
+    "vsce": "^2.15.0",
```
Comment (karthiknadig) [inline package.json:465]: You should be using `@vscode/vsce`

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    name: Tests
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: ${{ env.special-working-directory }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python: ['3.7', '3.8', '3.9', '3.10', '3.11-dev']
```
Comment (karthiknadig) [inline .github/workflows/pr-check.yml:48]:
```suggestion
        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
```

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+name: PR Validation
+
+on:
+  pull_request:
+
+env:
+  NODE_VERSION: 14.19.0
```
Comment (karthiknadig) [inline .github/workflows/pr-check.yml:7]:
```suggestion
  NODE_VERSION: 16.17.0
```

File: .github/workflows/push-check.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+on:
+  push:
+    branches:
+      - 'main'
+      - 'release'
+      - 'release/*'
+      - 'release-*'
+
+env:
+  NODE_VERSION: 14.19.0
```
Comment (karthiknadig) [inline .github/workflows/push-check.yml:12]:
```suggestion
  NODE_VERSION: 16.17.0
```

File: .github/workflows/push-check.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    name: Tests
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: ${{ env.special-working-directory }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python: ['3.7', '3.8', '3.9', '3.10']
```
Comment (karthiknadig) [inline .github/workflows/push-check.yml:53]:
```suggestion
        python: ['3.7', '3.8', '3.9', '3.10', '3.11']
```

File: noxfile.py [hunk 6]
```diff
@@ -47,6 +38,32 @@ def _update_pip_packages(session: nox.Session) -> None:
     )
 
 
+@nox.session()
+def lint(session: nox.Session) -> None:
```
Comment (karthiknadig) [inline noxfile.py:42]: For future, this can be used for linting the entire extension.

**Review by karthiknadig — CHANGES_REQUESTED** [review]

**Review by karthiknadig — APPROVED** [review]

File: .github/workflows/issue-labels.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+              github.rest.issues.addLabels({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.issue.number,
+                labels: ['triage-needed']
+              })
+            } else {
+              console.log('This issue already has a "needs __" label, do not add the "triage-needed" label.')
+            }
```
Comment (karrtikr) [inline .github/workflows/issue-labels.yml:39]: We should be using microsoft/vscode-github-triage-actions

**Review by karrtikr — COMMENTED** [review]
This PR is still using an older way to use Github actions, please look at the Python extension and update the code accordingly.

**Pull request merged**